From e8f1ca159d6ac3436d475977042327d1f423c181 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Sun, 26 Apr 2020 11:11:12 +0800
Subject: [PATCH] drm/rockchip: rk618: Avoid one div operation

Get dst->clock from sclk_rate before it changed to avoid one div
operation.

Change-Id: I33a51fd8b9cef5c3e73fc4db436fe8c1dcc57306
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rk618/rk618_scaler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rk618/rk618_scaler.c b/drivers/gpu/drm/rockchip/rk618/rk618_scaler.c
index 12343f14b6c2..c59da7bc4872 100644
--- a/drivers/gpu/drm/rockchip/rk618/rk618_scaler.c
+++ b/drivers/gpu/drm/rockchip/rk618/rk618_scaler.c
@@ -301,8 +301,8 @@ static void rk618_scaler_bridge_mode_set(struct drm_bridge *bridge,
 	sclk_rate = (u64)dclk_rate * dst->vdisplay * dst->htotal;
 	do_div(sclk_rate, src->vdisplay * src->htotal);
 	sclk_rate = div_u64(sclk_rate, 1000);
+	dst->clock = sclk_rate;
 	sclk_rate = sclk_rate * 1000;
-	dst->clock = div_u64(sclk_rate, 1000);
 	scl->bridge->driver_private = dst;
 
 	DRM_DEV_INFO(scl->dev, "src=%s, dst=%s\n", src->name, dst->name);
-- 
2.35.3

